homeassistant/components/anomaly/binary_sensor.py

```diff
@@ -82,20 +60,21 @@
 )
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {vol.Required(CONF_SENSORS): cv.schema_with_slug_keys(SENSOR_SCHEMA)}
+    {vol.Required(const.CONF_SENSORS): cv.schema_with_slug_keys(SENSOR_SCHEMA)}
 )
 
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
-    """Set up the anomaly sensors."""
-    await async_setup_reload_service(hass, DOMAIN, PLATFORMS)
+async def _async_create_entities(hass: HomeAssistant, config: OrderedDict):
+    """Create the template binary sensors."""
 
+    config_sensors = config.get(const.CONF_SENSORS, {})
     sensors = []
-    for device_id, device_config in config[CONF_SENSORS].items():
-        monitored_entity_id = device_config[ATTR_ENTITY_ID]
-        attribute = device_config.get(CONF_ATTRIBUTE)
-        device_class = device_config.get(CONF_DEVICE_CLASS)
-        friendly_name = device_config.get(ATTR_FRIENDLY_NAME, device_id)
+    for device_id, device_config in config_sensors.items():
+        entity_id = generate_entity_id(ENTITY_ID_FORMAT, device_id, hass=hass)
+        unique_id = device_config.get(const.CONF_UNIQUE_ID)
+        friendly_name = device_config.get(const.CONF_FRIENDLY_NAME)
```

Comment:
Remove this option and just set a default name based on device id. Users change change name in the GUI since we offer unique_id option.